### PR TITLE
OSInfo: fix implicit size_t conversion.

### PR DESCRIPTION
**Pull request by mkrautz**

### CODE REVIEW

File: src/OSInfo.cpp [hunk 1]
```diff
@@ -34,7 +34,10 @@ static QString regString(wchar_t *string, int size) {
 	// that the NUL is not included in the returned
 	// string.
 	const size_t adjustedSize = wcsnlen(string, static_cast<size_t>(size));
-	return QString::fromWCharArray(string, adjustedSize);
+	if (adjustedSize > INT_MAX) {
+		return QString();
```
Comment (hacst) [inline src/OSInfo.cpp]: This is unreachable. The return value of wcsnlen is <= size which is itself an int.

**Comment by hacst** [comment]
LGTM see comment.

File: src/OSInfo.cpp (hunk 1 above)
Comment (mkrautz) [inline src/OSInfo.cpp]: Very good point. Thank you, human static analyzer. :-)

**Comment by mkrautz** [comment]
Fixed. Will land.

**Pull request merged**